### DISCUSSION TITLE
useFirestore: How to call related doc by id in other collection after data return?

### DISCUSSION THREAD

**Discussion by manjumjn** [vueuse/vueuse#2353] [Q&A]
Hi guys i need help with `useFirestore` addon,

Let's say i have a data collection named `authors` and other collection named `books` where `book` doc has `authorId`, I'm retriewing data as below. How can i get the books by `id` returned by author collecton on each `author` document? 

```` js
const authors = useFirestore(dbquery, undefined, { autoDispose: false });
````

Just trying to find if there is a way to retriew it reactively which i missed.

**Comment by kiyopikko** [comment]
@manjumjn 

Are you talking about client side join? (I'm not good at English so cannot totally understand what you said...)

If yes, I use the child component to use `useFirestore`.

```vue
<!-- PostList.vue -->
<template>
  <PostListItem v-for="post in posts" :key="post.id" :user-ref="post.userRef" />
</template>

<script setup lang="ts">
const posts = useFirestore(collection(db, "posts"))
</script>
```

```vue
<!-- PostListItem.vue -->
<template>
  <div>{{ user }}</div>
</template>

<script setup lang="ts">
const user = useFirestore(props.userRef)
</script>
```
